spec/requests/vendor_api/make_an_offer_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+RSpec.describe 'Vendor API - POST /api/v1/applications/:id/offer', type: :request do
+  include VendorApiSpecHelpers
+
+  context 'successfully made a conditional offer' do
+    let(:application_choice) { create(:application_choice, provider_ucas_code: 'ABC') }
+    let(:request_body) do
+      {
+        "data": {
+          "conditions": [
+            'Completion of subject knowledge enhancement',
+            'Completion of professional skills test',
+          ],
+        },
+      }
+    end
+
+    before { post "/api/v1/applications/#{application_choice.id}/offer", params: request_body }
+
+    it 'returns a response that is valid according to the OpenAPI schema' do
+      expect(parsed_response).to be_valid_against_openapi_schema('SingleApplicationResponse')
+    end
+
+    it 'returns an application with a conditional offer' do
+      expect(parsed_response['data']['attributes']['status']).to eq('conditional_offer')
+      expect(parsed_response['data']['attributes']['offer']).to eq(
+        'conditions' => [
+          'Completion of subject knowledge enhancement',
+          'Completion of professional skills test',
+        ],
+      )
+    end
+  end
+
+  context 'successfully made an unconditional offer' do
+    let(:application_choice) { create(:application_choice, provider_ucas_code: 'ABC') }
+    let(:request_body) do
+      {}
+    end
+
+    before { post "/api/v1/applications/#{application_choice.id}/offer", params: request_body }
+
+    it 'returns a response that is valid according to the OpenAPI schema' do
+      expect(parsed_response).to be_valid_against_openapi_schema('SingleApplicationResponse')
+    end
+
+    it 'returns an application with an unconditional offer' do
+      expect(parsed_response['data']['attributes']['status']).to eq('unconditional_offer')
```

Comment:
I'm not sure, since this is an Enum, should we refer to it in a different way instead of just plain string? what is the alternative way?

Reply:
🤔 Not sure. It is an assertion on the request response. 
Is this not a string?

Reply:
ops sorry, does not matter, that is the response of the endpoint, it's not the model